bundles/mapping/mapmodule/MapModuleClass.ol.js

```diff
@@ -530,32 +531,173 @@ export class MapModule extends AbstractMapModule {
      *  (other components wont know that the map has moved, only use when chaining moves and
      *     wanting to notify at end of the chain for performance reasons or similar) (optional)
+     * @param {Object} options options, such as animation and duration
+     *     Usable animations: fly/pan/zoomPan
      * @return {Boolean} success
      */
-    centerMap (lonlat, zoom, suppressEnd) {
+    centerMap (lonlat, zoom, suppressEnd, options) {
+        const view = this.getMap().getView();
+        const animation = options && options.animation ? options.animation : '';
+        const duration = options && options.duration ? options.duration : 3000;
+        // If no animation, notify without duration
+        const notifyDuration = animation === '' ? 0 : duration;
+
         lonlat = this.normalizeLonLat(lonlat);
         if (!this.isValidLonLat(lonlat.lon, lonlat.lat)) {
             return false;
         }
-        this.getMap().getView().setCenter([lonlat.lon, lonlat.lat]);
+
         if (zoom === null || zoom === undefined) {
-            zoom = this.getMapZoom();
+            zoom = { type: 'zoom', value: this.getMapZoom() };
         }
-        this.getMap().getView().setZoom(zoom);
+        if (zoom === Number) {
+            zoom = { type: 'zoom', value: zoom };
+        }
+        const zoomValue = zoom.type === 'scale' ? view.getZoomForResolution(zoom.value) : zoom.value;
+
+        this._animateTo(lonlat, zoomValue, animation, duration);
+
         this.updateDomain();
         if (suppressEnd !== true) {
-            this.notifyMoveEnd();
+            setTimeout(() => {
+                this.notifyMoveEnd();
+            }, notifyDuration);
         }
         return true;
     }
+
+    /**
+     * @method tourMap
+     * Moves the map from point to point
+     * @param {Object[]} coordinates array of coordinates to move the map along
+     * @param {Object | Number} zoom absolute zoomlevel to set the map to
+     * @param {Object} options options, such as animation and duration
+     *     Usable animations: fly/pan/zoomPan
+     */
+    tourMap (coordinates, zoom, options) {
+        const view = this.getMap().getView();
+        const me = this;
+        const duration = options && options.duration ? options.duration : 3000;
+        const delayOption = options && options.delay ? options.delay : 750;
+        const animation = options && options.animation ? options.animation : '';
+
+        if (zoom === null || zoom === undefined) {
+            zoom = { type: 'zoom', value: this.getMapZoom() };
+        }
+
+        const zoomDefault = zoom.type === 'scale' ? view.getZoomForResolution(zoom.value) : zoom.value;
+        let index = -1;
+        let delay = 0;
+        let status = { steps: coordinates.length };
+
+        const next = (more) => {
+            if (index !== -1) {
+                // trigger if this wasn't the first call
+                me.notifyTourEvent(status, !more);
```

Comment:
Add calls to:

this.notifyMoveEnd();
this.updateDomain();

So components that update their UI based on map location refresh accordingly

Reply:
Scratch that. This https://github.com/oskariorg/oskari-frontend/blob/1.53.1/bundles/mapping/mapmodule/MapModuleClass.ol.js#L101 gets triggered with the animate method so the events and domain get updated.